obelix/idefix/services/google/gcalendar.go

```diff
@@ -0,0 +1,125 @@
+package google
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/Shopify/sarama"
+	redisv8 "github.com/go-redis/redis/v8"
+	"google.golang.org/api/calendar/v3"
+	"idefix/operator"
+	"idefix/producer"
+	"idefix/redis"
+	"log"
+	"time"
+)
+
+const (
+	ACTIVE = iota
+	NoActive
+)
+
+type GCalendar struct {
+	date string
+}
+
+func (gc *GCalendar) getEventsList(srv *calendar.Service) (*calendar.Events, error) {
+	t := time.Now().Format(time.RFC3339)
+	events, err := srv.Events.List("primary").
+		ShowDeleted(false).
+		SingleEvents(true).
+		TimeMin(t).
+		MaxResults(1).
+		OrderBy("startTime").
+		Do()
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve next event gcalendar: %w", err)
+	}
+
+	return events, nil
+}
+
+func (gc *GCalendar) Parse(srv *calendar.Service) error {
+	events, err := gc.getEventsList(srv)
+	if err != nil {
+		return fmt.Errorf("failed to get events list gcalendar: %w", err)
+	}
+
+	if len(events.Items) == 0 {
+		fmt.Println("No upcoming events found.")
```

Comment:
You can return then, that avoid you to do a useless else

Reply:
f6f17784c359d8bc4cf438bb23550f9b182e5830